src/App.tsx

```diff
@@ -1,16 +1,195 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoErrorNotification } from './components/TodoErrorNotification';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { SortType } from './types/SortType';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [sortType, setSortType] = useState(SortType.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [title, setTitle] = useState('');
+  const [isLoader, setIsLoader] = useState(false);
+  const [isLoaderCompletedTodo, setIsLoaderCompletedTodo] = useState(false);
+  const [isLoaderUnCompletedTodo, setIsLoaderUnCompletedTodo] = useState(false);
+  const [isDoubleClick, setIsDoubleClick] = useState(false);
```

Comment:
why do you need this `isDoubleClick `? and you use it many times. it's not necessary to check always. make it simpler. you have `onDoubleClick` react event, so make your logic simpler.